src/routes/safe/components/Apps/components/AppsList.test.tsx

```diff
@@ -0,0 +1,157 @@
+      {
+        id: 36,
+        url: 'https://apps.gnosis-safe.io/drain-safe',
+        name: 'Drain safe',
+        iconUrl: 'https://apps.gnosis-safe.io/drain-safe/logo.svg',
+        error: false,
+        description: 'Transfer all your assets in batch',
+        fetchStatus: 'SUCCESS',
+        chainIds: [4],
+        provider: null,
+      },
+      {
+        id: 3,
+        url: 'https://app.ens.domains',
+        name: 'ENS App',
+        iconUrl: 'https://app.ens.domains/android-chrome-144x144.png',
+        error: false,
+        description: 'Decentralised naming for wallets, websites, & more.',
+        fetchStatus: 'SUCCESS',
+        chainIds: [1, 4],
+        provider: null,
+      },
+      {
+        id: 14,
+        url: 'https://cloudflare-ipfs.com/ipfs/QmXLxxczMH4MBEYDeeN9zoiHDzVkeBmB5rBjA3UniPEFcA',
+        name: 'Synthetix',
+        iconUrl: 'https://cloudflare-ipfs.com/ipfs/QmXLxxczMH4MBEYDeeN9zoiHDzVkeBmB5rBjA3UniPEFcA/Synthetix.png',
+        error: false,
+        description: 'Trade synthetic assets on Ethereum',
+        fetchStatus: 'SUCCESS',
+        chainIds: [1, 4],
+        provider: null,
+      },
+
+      {
+        id: 24,
+        url: 'https://cloudflare-ipfs.com/ipfs/QmdVaZxDov4bVARScTLErQSRQoxgqtBad8anWuw3YPQHCs',
+        name: 'Transaction Builder',
+        iconUrl: 'https://cloudflare-ipfs.com/ipfs/QmdVaZxDov4bVARScTLErQSRQoxgqtBad8anWuw3YPQHCs/tx-builder.png',
+        error: false,
+        description: 'A Safe app to compose custom transactions',
+        fetchStatus: 'SUCCESS',
+        chainIds: [1, 4, 56, 100, 137, 246, 73799],
+        provider: null,
+      },
+      {
+        id: '0.44667970656294087',
+        url: 'http://localhost:3001',
+        name: 'unknown',
+        iconUrl: '/static/media/apps.70657d32.svg',
+        error: 'Failed to fetch app manifest',
+        description: '',
+        fetchStatus: 'ERROR',
+        disabled: false,
+        custom: true,
+      },
+    ],
+    removeApp: (appUrl: string) => {},
+    isLoading: false,
+  }),
+}))
+
+const customState = {
+  router: {
+    location: {
+      pathname: '/safes/0x75096d02718d1B56BEaE4273b178d34F6695F097/balances',
+    },
+    // had to include this because of checks in connected-react-router
+    // https://github.com/supasate/connected-react-router/issues/312#issuecomment-500968504
+    action: 'truthy',
+  },
+}
+
+describe('Safe Apps -> AppsList', () => {
+  it('Shows apps from the app list', () => {
+    render(<AppsList />, customState)
+
+    expect(screen.getByText('Compound')).toBeInTheDocument()
+  })
+
+  it("Doesn't show apps with unavailable manifest.json", () => {
+    render(<AppsList />, customState)
+
+    expect(screen.queryByText('unknown')).not.toBeInTheDocument()
+  })
+})
+
+describe('Safe Apps -> AppsList -> Search', () => {
+  it('Shows apps matching the search query', () => {
+    render(<AppsList />, customState)
+
+    const searchInput = screen.getByPlaceholderText('e.g Compound')
```

Comment:
I think it may be better to add a `data-testid` to the input instead, just in case we change the placeholder.

Reply:
You could even then create a getter function. You get the input a lot throughout these tests.

Reply:
> I think it may be better to add a data-testid to the input instead, just in case we change the placeholder.

No, it's not. Test Ids are the last resort when you cannot reach the element using attributes visible to the user—the whole philosophy of react-testing-library: tests that resemble user behavior. Users don't look up elements by test ids.

Also, test ids provide imagination of test security, if I change an element type from, let's say, input to a button but keep the test id, the test would still pass, but it should've not.

Reply:
> You could even then create a getter function. You get the input a lot throughout these tests.

I prefer to keep tests simple, and I don't think we should over-engineer them. When you read a text case, you understand what's inside and don't have to jump the file to find stuff defined outside the test case. If you'd have to change it, you can use CMD+Shift+L hotkey, and that's it.